cmd/generate/config/rules/gitlab.go

```diff
@@ -115,6 +115,23 @@ func GitlabPat() *config.Rule {
 	return utils.Validate(r, tps, fps)
 }
 
+func GitlabPatRoutable() *config.Rule {
+	r := config.Rule{
+		RuleID:      "gitlab-pat-routable",
+		Description: "Identified a GitLab Personal Access Token (routable), risking unauthorized access to GitLab repositories and codebase exposure.",
+		Regex:       regexp.MustCompile(`glpat-[0-9a-zA-Z_-]{27,300}\.[0-9a-z]{2}[0-9a-z]{7}`),
```

Comment:
I wonder why they did `[0-9a-z]{2}[0-9a-z]{7}` instead of `[0-9a-z]{9}`.

Reply:
I suppose it's to stay consistent with the [code](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/172708/diffs#a7e29018df78ada984f3110c2ea00eb20f0ef907_0_6) that generates the tokens:
```
    BASE64_PAYLOAD_LENGTH_HOLDER_BYTES = 2
    CRC_BYTES = 7
```

That way it's easier to spot which regex part matches which part of the token.

Reply:
Makes sense!